src/subscription_manager/cache.py

```diff
@@ -191,6 +195,86 @@ def update_check(self, uep, consumer_uuid, force=False):
             return 0  # No updates performed.
 
 
+class InMemoryCache(object):
+    """
+    Store, in-memory only, some values in a thread-safe way.
+    """
+
+    def __init__(self):
+        self._cache = {}
+        self._lock = threading.RLock()
+
+    def __enter__(self):
+        """
+        Allow this cache object to be used as a context manager.
+        This only acquires the lock. The lock will be released in __exit__.
+        :return:
+        """
+        self._lock.acquire()
+        log.debug("{self}: lock acquired".format(self=self))
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """
+        Release the lock on exit
+        :param exc_type:
+        :param exc_val:
+        :param exc_tb:
+        :return:
+        """
+        self._lock.release()
+        log.debug("{self}: lock released".format(self=self))
+
+    def write_cache(self, key, data):
+        """
+        Overwrite the contents of the cache with data.
+        :param data: the contents
+        :return:
+        """
+        self._lock.acquire()
+        self._cache[key] = copy.deepcopy(data)
+        self._lock.release()
+        return copy.deepcopy(self._safe_get(key))
+
+    def read_cache_only(self, key=None):
+        """
+        Return the contents of the cache
+        :return:
+        """
+        self._lock.acquire()
+        if key is not None:
+            data = copy.deepcopy(self._safe_get(key))
+        else:
+            data = copy.deepcopy(self._cache)
+        self._lock.release()
+        return data
+
+    def _safe_get(self, key):
+        """
+        Try to read the key from the _cache dict, this could fail if the _cache is some how
+        modified to something that doesn't have a "get" method
+        :param key:
+        :return:
+        """
+        try:
+            data = self._cache.get(key, None)
+            return data
+        except AttributeError:
+            log.debug('Looks like the _cache was modified by something else, it was not a dict')
+            return None
+
+
+class ReadThroughInMemoryCache(InMemoryCache):
```

Comment:
We should remove this b/c it is not used (and/or shouldn't be used as of yet). Please either remove or let me know why we should keep it.

Reply:
removed